packages/element-lib/src/sidetree-v2/func.js

```diff
@@ -0,0 +1,136 @@
+  const decodedOperation = decodeJson(op);
+  const operationHash = payloadToHash(decodedOperation.payload);
+  const decodedOperationPayload = decodeJson(decodedOperation.payload);
+  return {
+    operationHash,
+    decodedOperation,
+    decodedOperationPayload,
+  };
+});
+
+const syncTransaction = async (sidetree, transaction) => {
+  const { transactionNumber } = transaction;
+  const anchorFile = await sidetree.storage.read(transaction.anchorFileHash);
+  if (!schema.validator.isValid(anchorFile, schema.schemas.sidetreeAnchorFile)) {
+    // TODO
+    // console.warn('anchorFile not valid', anchorFile);
+    return null;
+  }
+  const batchFile = await sidetree.storage.read(anchorFile.batchFileHash);
+  if (!schema.validator.isValid(batchFile, schema.schemas.sidetreeBatchFile)) {
+    // console.warn('batch file not valid', anchorFile);
+    return null;
+  }
+  const operations = batchFileToOperations(batchFile);
+  const operationsByDidUniqueSuffixes = operations.map((operation) => {
+    const { decodedOperationPayload } = operation;
+    const didUniqueSuffix = decodedOperationPayload.didUniqueSuffix
+      ? decodedOperationPayload.didUniqueSuffix
+      : payloadToHash(decodedOperationPayload);
+    return {
+      type: didUniqueSuffix,
+      didUniqueSuffix,
+      transactionNumber,
+      ...operation,
+    };
+  });
+  const writeOperationToCache = op => sidetree.db.write(`operation:${op.operationHash}`, op);
+  return executeSequentially(writeOperationToCache, operationsByDidUniqueSuffixes)
+    .then(() => {
+      return sidetree.db.write(`transaction:${transaction.transactionNumber}`, {
+        type: 'transaction',
+        transactionNumber: transaction.transactionNumber,
+      });
+    }).catch((error) => {
+      console.log(error);
+      // https://stackoverflow.com/questions/18391212/is-it-not-possible-to-stringify-an-error-using-json-stringify
+      const stringifiedError = JSON.stringify(error, Object.getOwnPropertyNames(error));
+      return sidetree.db.write(`transaction:${transaction.transactionNumber}`, {
+        type: 'transaction',
+        transactionNumber: transaction.transactionNumber,
+        error: stringifiedError,
+      });
+    });
+};
+
+const isTransactionValid = (transaction) => {
+  const valid = schema.validator.isValid(transaction, schema.schemas.sidetreeTransaction);
+  if (!valid) {
+    console.warn('bad transaction', transaction);
+  }
+  return valid;
+};
+
+
+const signEncodedPayload = (encodedPayload, privateKeyHex) => {
+  const toBeSigned = `.${encodedPayload}`;
+  const hash = crypto
+    .createHash('sha256')
+    .update(Buffer.from(toBeSigned))
+    .digest();
+  const privateKeyBuffer = Buffer.from(privateKeyHex, 'hex');
+  const signatureObject = secp256k1.sign(hash, privateKeyBuffer);
+  const signature = base64url.encode(signatureObject.signature);
+  return signature;
+};
+
+
+module.exports = {
```

Comment:
I love this minimal interface... once we get some feedback on these changes, I would like to consider moving this stuff to element-core / typescript.